From 9547e5502cdfed215efd8c234e81b14d714316e7 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Sun, 23 Feb 2020 20:29:45 +0800
Subject: [PATCH 28/37] MLK-23233-1 arm64: dts: refine pcie dts and add
 pcieax2pciebx1 usecase

commit 68b44ec06a904cd2dc95d315b705c7469e65c721 from
https://source.codeaurora.org/external/imx/linux-imx.git

Different usecase maybe used by customer, add the PCIEA two lanes and
PCIEB one lane usecase into fsl-imx8qm-pcieax2pciebx1.dts.
Refine the PCIE dts nodes, add the requrired HSIO peripheral clocks for
different consumers.
PCIEB has one more PER clock, since the PCIEA CSR register would be
configuired when PCIEB is initialized.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Reviewed-by: Fugang Duan <fugang.duan@nxp.com>
(cherry picked from commit f30146a26201e8ff84d6f8d7e0debde957c4db59)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  1 +
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi |  5 +-
 .../freescale/fsl-imx8qm-pcieax2pciebx1.dts   | 56 +++++++++++++++++++
 3 files changed, 60 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-pcieax2pciebx1.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 348a3993de90..ee7122add0d5 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -48,6 +48,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-mek-domu-dpu1-hdmi.dtb \
 				 fsl-imx8qm-mek-root.dtb \
 				 fsl-imx8qm-mek-inmate.dtb \
+				 fsl-imx8qm-pcieax2pciebx1.dtb \
 				 fsl-imx8qm-lpddr4-arm2-dp.dtb \
 				 fsl-imx8qm-lpddr4-arm2-hdmi.dtb \
 				 fsl-imx8qm-lpddr4-arm2-hdmi-in.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index 433c2cd1d113..920ddab445dc 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -4152,7 +4152,7 @@
 					 <0 0 0 2 &gic 0 74 4>,
 					 <0 0 0 3 &gic 0 75 4>,
 					 <0 0 0 4 &gic 0 76 4>;
-			power-domains = <&pd_pcie1>;
+			power-domains = <&pd_pcie0>;
 			fsl,max-link-speed = <3>;
 			hsio-cfg = <PCIEAX1PCIEBX1SATA>;
 			ctrl-id = <0>; /* pciea */
@@ -4186,10 +4186,11 @@
 				 <&clk IMX8QM_HSIO_PCIE_B_SLV_AXI_CLK>,
 				 <&clk IMX8QM_HSIO_PHY_X2_PCLK_1>,
 				 <&clk IMX8QM_HSIO_PCIE_X1_PER_CLK>,
+				 <&clk IMX8QM_HSIO_PCIE_X2_PER_CLK>,
 				 <&clk IMX8QM_HSIO_PCIE_B_DBI_AXI_CLK>,
 				 <&clk IMX8QM_HSIO_PHY_X2_PER_CLK>,
 				 <&clk IMX8QM_HSIO_MISC_PER_CLK>;
-			clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_per",
+			clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_per", "pciex2_per",
 				"pcie_inbound_axi", "phy_per", "misc_per";
 
 			interrupt-map-mask = <0 0 0 0x7>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-pcieax2pciebx1.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-pcieax2pciebx1.dts
new file mode 100644
index 000000000000..4f283816843d
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-pcieax2pciebx1.dts
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright 2020 NXP
+ */
+
+
+#include "fsl-imx8qm-mek.dts"
+
+/*
+ * Add the PCIeA x2 lanes and PCIeB x1 lane usecase
+ * hsio-cfg = <PCIEAX2PCIEBX1>
+ * NOTE: In this case, the HSIO nodes contained
+ * hsio-cfg = <PCIEAX1PCIEBX1SATA> would be re-configured.
+ */
+&pciea{
+	ext_osc = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pciea>;
+	disable-gpio = <&gpio4 9 GPIO_ACTIVE_LOW>;
+	reset-gpio = <&gpio4 29 GPIO_ACTIVE_LOW>;
+	clkreq-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
+	epdev_on-supply = <&epdev_on>;
+	num-lanes = <2>;
+	clocks = <&clk IMX8QM_HSIO_PCIE_A_MSTR_AXI_CLK>,
+		 <&clk IMX8QM_HSIO_PCIE_A_SLV_AXI_CLK>,
+		 <&clk IMX8QM_HSIO_PHY_X2_PCLK_0>,
+		 <&clk IMX8QM_HSIO_PCIE_X2_PER_CLK>,
+		 <&clk IMX8QM_HSIO_PCIE_A_DBI_AXI_CLK>,
+		 <&clk IMX8QM_HSIO_PHY_X2_PER_CLK>,
+		 <&clk IMX8QM_HSIO_MISC_PER_CLK>;
+	clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_per",
+			"pcie_inbound_axi", "phy_per", "misc_per";
+	hsio-cfg = <PCIEAX2PCIEBX1>;
+	status = "okay";
+};
+
+&pcieb{
+	ext_osc = <1>;
+	clocks = <&clk IMX8QM_HSIO_PCIE_B_MSTR_AXI_CLK>,
+		 <&clk IMX8QM_HSIO_PCIE_B_SLV_AXI_CLK>,
+		 <&clk IMX8QM_HSIO_PHY_X1_PCLK>,
+		 <&clk IMX8QM_HSIO_PCIE_X1_PER_CLK>,
+		 <&clk IMX8QM_HSIO_PCIE_X2_PER_CLK>,
+		 <&clk IMX8QM_HSIO_PCIE_B_DBI_AXI_CLK>,
+		 <&clk IMX8QM_HSIO_PHY_X1_PER_CLK>,
+		 <&clk IMX8QM_HSIO_MISC_PER_CLK>;
+	clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_per", "pciex2_per",
+			"pcie_inbound_axi", "phy_per", "misc_per";
+	power-domains = <&pd_pcie1>;
+	hsio-cfg = <PCIEAX2PCIEBX1>;
+	status = "okay";
+};
+
+&sata {
+	status = "disabled";
+};
-- 
2.17.1

